src/dmd/console.d

```diff
@@ -166,15 +195,22 @@ struct Console
       public:
 
         @property FILE* fp() { return _fp; }
-
-        static Console* create(FILE* fp)
+        /**
+         Tries to detect whether DMD has been invoked from a terminal.
+         Returns: `true` if a terminal has been detect, `false` otherwise
+         */
+        static bool detectTerminal()
         {
             import core.stdc.stdlib : getenv;
             const(char)* term = getenv("TERM");
             import core.stdc.string : strcmp;
             if (!(isatty(STDERR_FILENO) && term && term[0] && 0 != strcmp(term, "dumb")))
-                return null;
+                return false;
+            return true;
```

Comment:
This logic would be simpler as:
```
return isatty(STDERR_FILENO) && term && term[0] && 0 != strcmp(term, "dumb"));
```

Reply:
Should it use a D string instead?

Reply:
`getenv("TERM")` return a C string, so not really worth it.